crates/bevy_ecs/src/event.rs

```diff
@@ -46,6 +52,32 @@ impl<E: Event> fmt::Debug for EventId<E> {
     }
 }
 
+impl<E: Event> PartialEq for EventId<E> {
+    fn eq(&self, other: &Self) -> bool {
+        self.id == other.id
+    }
+}
+
+impl<E: Event> Eq for EventId<E> {}
+
+impl<E: Event> PartialOrd for EventId<E> {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.id.cmp(&other.id))
```

Comment:
```suggestion
        Some(self.cmp(&other))
```
To ensure consistency between the two methods

Reply:
Thanks for the feedback. Fixed it.

I had just copy pasted that part from somewhere else in the codebase. Maybe a repo wide check for this should be done at some point?